wave_lang/kernel/wave/utils/general_utils.py

```
@@ -71,6 +71,20 @@ def get_default_scheduling_params() -> dict[IndexSymbol, Any]:
     }
 
 
+def linearize_index(expr_list: dict[IndexExpr, IndexSequence], strides: list[int]):
+    assert len(expr_list) == len(strides)
+    linear_expr = None
+    for expr, stride in zip(expr_list.values(), strides):
+        if not isinstance(expr, IndexSequence):
+            return None
```

We should probably raise an error here instead of returning `None`.